### DISCUSSION TITLE
Unable to search GH code without login

### DISCUSSION THREAD

**Discussion by gh-at-sqh** [community/community#58510] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

Previously, I don't know how many moons ago (can't be more than a few), I was able to search github code anonymously, from the little search box at the top. The default search looked through the current browsed repo's code.

Search result would be all occurrences of word(s) in any searchable file in the repo master. It was pretty useful despite its shortcomings. There are innumerable benefits to repo text search as I'm sure we can all agree.

Today (6/2023) search appears only to look at a repo's "issues". Selecting the "code" tab in the result now takes me to a login page. How annoying. Is that the intended behavior with the MS buyout/devolution?

Am I no longer an important enough individual to provide search results unless some facet of my identity is known? You've got my IP address, what more do you want? You could (and no doubt do) surmise my identity (to 98.9% probability, but without explicit consent) just with that. Or do you just want to know how many different identities I can fabricate? At least 2^31 - 1. For every one I have to "verify" with "validation" nonsense through the email wash. Must I be burdened so?

Now I must also remember to do my best to never accidentally or on purpose associate a GH account with an MS one. Otherwise I'll (by default) explicitly consent (to appease the lawyers) to be data-mined up the arse-end, something I wouldn't agree to even in prison.

The login requirement serves a customer no benefit that I can see. It used to be sung (quite popularly) that "**[if you mind your own business you won't be mindin' mine](https://www.youtube.com/watch?v=YlBusgSKiHM)**." Respectful (and moral) companies have an utterly strict policy to mind their own business, and the policy is instituted and enforced top-down and bottom-up without gainsaying.

I'd like to see GH continue to grow, prosper, and expand, but few poisons make good fertilizer. The fact is, the very best you can do is plant, water, feed. You cannot and do not cause the increase.

If I am mistaken or behind the times and updates, please let me know so I can go back to anonymous browsing and anonymous search. Please reveal the secret if it still exists. Thank you.

---
[The original version may be more to your liking](https://www.youtube.com/watch?v=RZH2bmbUTl4)

**Comment by look** [comment, marked as answer]
As of June 7, 2023, [all code search queries now require being logged in](https://github.blog/changelog/2023-06-07-code-search-now-requires-login/). Previously, repo-scoped code search queries could be performed without logging in. We removed that as part of our migration to our new [code search system](https://github.blog/2023-05-08-github-code-search-is-generally-available/), which requires being authenticated.

**Reply by qkaseman** [reply]
What is the technical reason why logging in is now required? Nothing in either link you provided gave any justification for this change.

**Reply by look** [reply]
As I said before, the search engine does not support authenticated queries. We'd have to implement that functionality. However, allowing unauthenticated searches is an abuse vector that was widely exploited, so we have chosen not to do so.

**Reply by Ark-kun** [reply]
@look 
>the search engine does not support authenticated queries. 

The search engine is an inanimate object it cannot decide what to do. It cannot decide to change. Let's use proper English language with proper active voice:

"Our search engine has previously supported code search for all users. However, recently our team decided to make a change to prohibit search requests from unauthorized users."

It's more honest this way, right?

**Reply by Ark-kun** [reply]
Given that GitHub now logs off every day or so on mobile, the GitHub search is essentially dead.

I wonder how many human hours of planning and development it took to kill one of the core and most important GitHub features - Search.

**Reply by Ark-kun** [reply]
This also kills many other applications. With new changes, if an open-source web app needs to search for certain labels or files on behalf of it's user, the app must now request the user's authentication with GitHub. The user must now risk their account, risk their PII just to let an app search public GitHub repos. Something that has always been possible.
